### PR TITLE
List verification

### PR DESCRIPTION
**Pull request by andreistefanescu**

### CODE REVIEW

**Comment by andreistefanescu** [comment]
@cos and @daejunpark please review

File: java-backend/src/main/java/org/kframework/backend/java/symbolic/ConjunctiveFormula.java [hunk 1]
```diff
@@ -94,6 +94,7 @@ public ConjunctiveFormula(
             GlobalContext global) {
         super(Kind.KITEM);
 
+        assert !equalities.stream().anyMatch(e -> e.leftHandSide() instanceof BuiltinList && ((BuiltinList) e.leftHandSide()).isEmpty() && e.rightHandSide() instanceof BuiltinList && !((BuiltinList) e.rightHandSide()).isEmpty() && ((BuiltinList) e.rightHandSide()).isElement(0));
```
Comment (cos) [inline java-backend/src/main/java/org/kframework/backend/java/symbolic/ConjunctiveFormula.java:98]: kind of hard to read -- add a comment

File: java-backend/src/main/java/org/kframework/backend/java/symbolic/FastRuleMatcher.java [hunk 2]
```diff
@@ -375,8 +374,15 @@ private BitSet matchAssoc(BuiltinList subject, int subjectIndex, BuiltinList pat
...
-
         BuiltinList.ElementTailSplit patternElementTailSplit = pattern.splitElementTail(patternIndex, ruleCount);
+        if (subjectIndex < subject.size() && !subject.isElement(subjectIndex) && !ruleMask.subset(patternElementTailSplit.tailMask)) {
+            return addUnification(
+                    subject.range(subjectIndex, subject.size()),
+                    pattern.range(patternIndex, pattern.size()),
+                    ruleMask,
+                    subject instanceof BuiltinList.SingletonBuiltinList ? path : path.$colon$colon(Pair.of(subjectIndex, subject.size())));
+        }
+
```
Comment (cos) [inline java-backend/src/main/java/org/kframework/backend/java/symbolic/FastRuleMatcher.java:389]: what gets unified?

**Review by cos — APPROVED** [review]
looks good to me

**Comment by daejunpark** [comment]
@andreistefanescu could you give a quick summary of this change?

**Comment by andreistefanescu** [comment]
It is collection of patches that enables the verification to work on the list examples again. There is no unifying theme. Mostly changes that have to do with differences between kore and the old kil, and with handling assoc matching/unification.

**Comment by daejunpark** [comment]
Thanks, lgtm.

**Pull request merged**